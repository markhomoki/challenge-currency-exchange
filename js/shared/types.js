export const EUR = 'EUR';
export const GBP = 'GBP';
export const USD = 'USD';

export const CURRENCIES = {
	[EUR]: '€',
	[GBP]: '£',
	[USD]: '$',
};
