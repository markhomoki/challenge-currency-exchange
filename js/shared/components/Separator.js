import React from 'react';
import PropTypes from 'prop-types';
import { CURRENCIES, EUR, GBP, USD } from 'app/types';

export default class Separator extends React.PureComponent {

	static propTypes = {
		fromCurrency: PropTypes.oneOf([EUR, GBP, USD]).isRequired,
		onSwap: PropTypes.func.isRequired,
		rate: PropTypes.number.isRequired,
		toCurrency: PropTypes.oneOf([EUR, GBP, USD]).isRequired,
	}

	renderSymbol = (currency) => {
		return CURRENCIES[currency] || '';
	}

	render() {
		const {
			fromCurrency,
			onSwap,
			rate,
			toCurrency,
		} = this.props;

		const fromSymbol = this.renderSymbol(fromCurrency);
		const toSymbol = this.renderSymbol(toCurrency);
		const pairValue = 1 * rate;

		return (
			<div className="separator">
				<button type="button" className="btn-exchange" onClick={onSwap}>
					<i className="ion-arrow-swap" />
				</button>
				<div className="rate-wrapper">
					<i className="ion-arrow-graph-up-right" />
					<span className="rate">{fromSymbol}1 = {toSymbol}{pairValue.toFixed(4)}</span>
				</div>
			</div>
		);
	}

}
