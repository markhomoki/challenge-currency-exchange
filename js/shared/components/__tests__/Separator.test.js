import React from 'react';
import renderer from 'react-test-renderer';
import { GBP, USD } from 'app/types';
import Separator from '../Separator';

describe('Separator', () => {
	const element = (
		<Separator
			fromCurrency={USD}
			onSwap={() => {}}
			rate={1.2}
			toCurrency={GBP}
		/>
	);

	it('renders correctly', () => {
		const tree = renderer.create(element).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
