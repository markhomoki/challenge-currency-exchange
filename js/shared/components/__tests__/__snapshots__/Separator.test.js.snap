// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator renders correctly 1`] = `
<div
  className="separator"
>
  <button
    className="btn-exchange"
    onClick={[Function]}
    type="button"
  >
    <i
      className="ion-arrow-swap"
    />
  </button>
  <div
    className="rate-wrapper"
  >
    <i
      className="ion-arrow-graph-up-right"
    />
    <span
      className="rate"
    >
      $
      1 = 
      Â£
      1.2000
    </span>
  </div>
</div>
`;
